sympy/core/equation.py

```diff
@@ -0,0 +1,359 @@
+    Integral(a*c, b)=b**2/(2*c)
+    >>> # This is duplicated by the convenience function self.integ 
+    >>> q.integ(b)                                                              
+    Integral(a*c, b)=b**2/(2*c)
+
+    SymPy's solvers do not understand these equations. They expect an expression that the solver assumes = 0.
+    Thus to use the solver the equation must be rearranged so that all non-zero symbols are on one side. Then
+    just the non-zero symbolic side is passed to ``solve()``.
+    >>> t2 = t-t.rhs
+    >>> t2
+    a-b/c=0
+    >>> solve(t2.lhs,c)
+    [b/a]
+    """
+
+    def __new__(cls, lhs, rhs, relop='='):
+        if not(relop == '='):
+           raise NotImplementedError('"=" is the only relational operator presently supported in Equations.')
+        lhs = _sympify(lhs)
+        rhs = _sympify(rhs)
+        if (lhs.is_number) and (rhs.is_number) and not (lhs == rhs):
+           print('WARNING: did your really mean to define unequal numbers as equal? ' + str(lhs) +'='+ str(rhs))
+        return super().__new__(cls, lhs, rhs, relop)
+
+    @property
+    def lhs(self):
+        """
+        Returns the lhs of the equation.
+        """
+        return self.args[0]
+
+    @property
+    def rhs(self):
+        """
+        Returns the rhs of the equation.
+        """
+        return self.args[1]
+    
+    @property
+    def relop(self):
+        """
+        Returns the string representing the relationship operator.
+        """
+        return self.args[2]
+
+    def as_Boolean(self):
+        """
+        Converts the equation to an Equality.
+        """
+        from .relational import Equality
+        return Equality(self.lhs, self.rhs)
+
+    @property
+    def reversed(self):
+        """
+        Swaps the lhs and the rhs.
+        """
+        return Equation(self.rhs, self.lhs)
+    
+    @property
+    def free_symbols(self):
+        ret =self.lhs.free_symbols 
+        ret.update(self.rhs.free_symbols)
+        return ret
+
+    def _applyfunc(self, func, *args, **kwargs):
+        # Assume if the expression has an attribute of name `func` that should override any general function
+        # Because there are name conflicts (e.g. `transpose` and `Matrix.transpose`) if either rhs or lhs
+        # has the attribute we will try to apply it to both. This will raise an error if both sides do
+        # not support the operation.
+
+        side=kwargs.pop('Eqn_apply_side',None)
+        if (side=='both'):
+            if (hasattr(self.lhs,str(func))) or (hasattr(self.rhs,str(func))):
+                return Equation(getattr(self.lhs,str(func))(*args, **kwargs),getattr(self.rhs,str(func))(*args, **kwargs))
+            else:
+                return Equation(func(self.lhs, *args, **kwargs), func(self.rhs, *args, **kwargs))
+        elif (side == 'lhs'):
+            if (hasattr(self.lhs,str(func))):
+                return Equation(getattr(self.lhs,str(func))(*args, **kwargs),self.rhs)
```

Comment:
I'm not so keen on this mechanism. I understand the motivation but I think maybe there's better way. Not all methods have corresponding functions which would make this awkward to use

Reply:
I am worried about it too, but do not think we should be writing overrides for each function. As I have been digging into the code more, I think there may need to be some decisions made about a consistent way of doing things like this. Or am I overlooking an already existing mechanism? There seem to be a number of different override mechanisms in the code. For example, the code includes the special `_eval_XXX` for things like integration and differentiation and other mechanisms for other operations.

Reply:
There are lots of different mechanisms around but none that would work for this case. I have an idea though but it will take a little time to mock up...

Reply:
Not applicable anymore as a different mechanism is now used to accommodate multiple requested syntaxes. See [commit 57ef7ab](https://github.com/sympy/sympy/pull/19479/commits/57ef7ab1cd42bebd2bbe5ed8b90c6c413b192b44) and [commit e9044ac](https://github.com/sympy/sympy/pull/19479/commits/e9044ac7fd52f35814749a0af75ddea49a499145).